### PR TITLE
test: net: igmp: Add IGMP query test

### PR DESCRIPTION
**Pull request by IVandeVeire**
Add functionality to test the IGMP query handling. This test should prevent the IGMP logic from sending IGMPv3 messages to IGMPv2 queriers. The test does now also check backwards compatability when IGMPv3 queries are received but only IGMPv2 is enabled.

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]

File: tests/net/igmp/src/main.c [hunk 1]
```diff
@@ -559,4 +607,58 @@ ZTEST_USER(net_igmp, test_socket_catch_join_with_index)
...
+	/* Joining group first to get reply on query*/
+	join_group();
+
+	is_igmpv2_query_sent = !is_imgpv3;
+	is_igmpv3_query_sent = is_imgpv3;
+
+	pkt = prepare_igmp_query(net_iface, is_imgpv3);
+	if (!pkt) {
+		zassert_true(false, "IGMPv2 query packet prep failed");
+	}
```
Comment (pdgendt) [inline tests/net/igmp/src/main.c:629]:
```suggestion
	zassert_not_null(pkt, "IGMPv2 query packet prep failed");
```

File: tests/net/igmp/src/main.c [hunk 2]
```diff
@@ -559,4 +607,58 @@ ZTEST_USER(net_igmp, test_socket_catch_join_with_index)
...
+
+	if (net_ipv4_input(pkt, false)) {
+		net_pkt_unref(pkt);
+		zassert_true(false, "Failed to send");
+	}
```
Comment (pdgendt) [inline tests/net/igmp/src/main.c:634]: It's already unexpected to fail, asserting would abort anyhow.

```suggestion
	zassert_equal(net_ipv4_input(pkt, false), NET_OK, "Failed to send");
```

File: tests/net/igmp/src/main.c [hunk 3]
```diff
@@ -559,4 +607,58 @@ ZTEST_USER(net_igmp, test_socket_catch_join_with_index)
...
+
+	if (k_sem_take(&wait_data, K_MSEC(WAIT_TIME))) {
+		zassert_true(0, "Timeout while waiting query event");
+	}
```
Comment (pdgendt) [inline tests/net/igmp/src/main.c:638]:
```suggestion
	zassert_ok(k_sem_take(&wait_data, K_MSEC(WAIT_TIME)), "Timeout while waiting query event");
```

File: tests/net/igmp/src/main.c [hunk 4]
```diff
@@ -559,4 +607,58 @@ ZTEST_USER(net_igmp, test_socket_catch_join_with_index)
...
+
+	if (!is_report_sent) {
+		zassert_true(0, "Did not catch query event");
+	}
```
Comment (pdgendt) [inline tests/net/igmp/src/main.c:642]:
```suggestion
	zassert_true(is_report_sent, "Did not catch query event");
```

File: tests/net/igmp/src/main.c [hunk 5]
```diff
@@ -559,4 +607,58 @@ ZTEST_USER(net_igmp, test_socket_catch_join_with_index)
...
+
+	if (!is_join_msg_ok) {
+		zassert_true(0, "Join msg invalid");
+	}
```
Comment (pdgendt) [inline tests/net/igmp/src/main.c:646]:
```suggestion
	zassert_true(is_join_msg_ok, "Join msg invalid");
```

**Review by pdgendt — CHANGES_REQUESTED** [review]
Some checks need to be fixed.

**Review by pdgendt — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**